translib/transformer/xfmr_testxfmr_callbacks.go

```diff
@@ -0,0 +1,713 @@
+	oc_testSetType := findInMap(TEST_SET_TYPE_MAP, data[TEST_SET_TABLE][inParams.key].Field[TEST_SET_TYPE])
+	n, err := strconv.ParseInt(oc_testSetType, 10, 64)
+	if n == int64(ocbinds.OpenconfigTestXfmr_TEST_SET_TYPE_TEST_SET_IPV4) {
+		result[TEST_SET_TYPE] = "TEST_SET_IPV4"
+	} else {
+		result[TEST_SET_TYPE] = "TEST_SET_IPV6"
+	}
+	return result, err
+}
+
+var DbToYang_test_set_description_field_xfmr FieldXfmrDbtoYang = func(inParams XfmrParams) (map[string]interface{}, error) {
+	result := make(map[string]interface{})
+	var err error
+	data := (*inParams.dbDataMap)[inParams.curDb]
+	log.Info("DbToYang_test_set_description_field_xfmr", data, inParams.ygRoot)
+	oc_testSetDesc := strings.SplitN(data[TEST_SET_TABLE][inParams.key].Field[TEST_SET_DESC], "Description : ", 2)[1]
+	result[TEST_SET_DESC] = oc_testSetDesc
+	log.Info("DbToYang_test_set_description_field_xfmr returning :", result)
+	return result, err
+}
+
+var YangToDb_test_set_description_field_xfmr FieldXfmrYangToDb = func(inParams XfmrParams) (map[string]string, error) {
+	var err error
+	var testSetDesc string
+	res_map := make(map[string]string)
+	log.Info("YangToDb_test_set_description_field_xfmr called")
+	if inParams.param == nil {
+		return res_map, err
+	}
+
+	testSetDescPtr, _ := inParams.param.(*string)
+	if testSetDescPtr != nil {
+		testSetDesc = *testSetDescPtr
+	}
+	res_map[TEST_SET_DESC] = "Description : " + testSetDesc
+	log.Info("YangToDb_test_set_description_field_xfmr: ", res_map)
+	return res_map, err
+}
+
+var YangToDb_test_port_bindings_xfmr SubTreeXfmrYangToDb = func(inParams XfmrParams) (map[string]map[string]db.Value, error) {
+	var err error
+	res_map := make(map[string]map[string]db.Value)
+	testSetTableMap := make(map[string]db.Value)
+	testSetTableMapNew := make(map[string]db.Value)
+	log.Info("YangToDb_test_port_bindings_xfmr: ", inParams.ygRoot, inParams.uri)
+
+	testXfmrObj := getTestSetRoot(inParams.ygRoot)
+	if testXfmrObj.Interfaces == nil {
+		return res_map, err
+	}
+
+	testSetTs := &db.TableSpec{Name: TEST_SET_TABLE}
+	testSetKeys, err := inParams.d.GetKeys(testSetTs)
+	if err != nil {
+		return res_map, err
+	}
+
+	for key := range testSetKeys {
+		testSetEntry, err := inParams.d.GetEntry(testSetTs, testSetKeys[key])
+		if err != nil {
+			return res_map, err
+		}
+		testSetTableMap[(testSetKeys[key].Get(0))] = testSetEntry
+	}
+
+	testSetInterfacesMap := make(map[string][]string)
+	for intfId, _ := range testXfmrObj.Interfaces.Interface {
+		intf := testXfmrObj.Interfaces.Interface[intfId]
+		if intf != nil {
+			if intf.IngressTestSets != nil && len(intf.IngressTestSets.IngressTestSet) > 0 {
+				for inTestSetKey, _ := range intf.IngressTestSets.IngressTestSet {
+					testSetName := getTestSetKeyStrFromOCKey(inTestSetKey.SetName, inTestSetKey.Type)
+					testSetInterfacesMap[testSetName] = append(testSetInterfacesMap[testSetName], *intf.Id)
+					_, ok := testSetTableMap[testSetName]
+					if !ok {
+						if inParams.oper == DELETE {
```

Comment:
nit: To reduce nesting, consider changing this to
```
if !ok && inParams.oper == DELETE {
   ...
}
```

Reply:
Addressed

Reply:
This was not changed. Did you forget to push your changes?

Reply:
Its already changed present in this [commit](https://github.com/sonic-net/sonic-mgmt-common/pull/81/commits/f83ddd2be2d045b3f24b5d75be739bc090773f39#diff-09e9f324d51d3fa405fdc481a9afc9ff694570eb13ecc26935e5973d7fa8ca14R485)